src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReference.tsx

```diff
@@ -0,0 +1,101 @@
+import { makeStyles } from "@material-ui/core";
+import useClipboard from "@saleor/hooks/useClipboard";
+import { CopyIcon, Pill } from "@saleor/macaw-ui";
+import React, { useEffect, useRef, useState } from "react";
+import { useIntl } from "react-intl";
+
+import { messages } from "../messages";
+
+const useStyles = makeStyles(
+  {
+    pill: {
+      willChange: "contents",
+    },
+  },
+  {
+    name: "PspReference",
+  },
+);
+
+enum RenderStep {
+  initial,
+  checkedMinimal,
+  final,
+}
```

Comment:
**nitpick**: I'd use union type here

Reply:
@Droniu why should I use a union type here? 🤔

Reply:
It doesn't matter that much, it's a nitpick. Union types are generally easier to work with than enums (unique values, mixed types, subtyping) however both provide type safety which is the main reason for using them.